### PR TITLE
Feature/homework4

### PR DESCRIPTION
**Pull request by SonaArshakyan**
Implemented the interaction between Catalog and Basket services via a message broker (Azure Service Bus).
The message will be send to Azure Service Bus when Product will be updated from Catalog Service and BasketItemUpdater 
 from basket Service as BackgroundService will be responsible to update  the basket item in Basket Service side.

### CODE REVIEW

File: Homework4/AzureServiceBusMessageBroker/Basket Service/Interfaces/Services/BasketItemUpdater.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿using AzureServiceBusMessageBroker.MessageModels;
+using AzureServiceBusMessageBroker.Services;
+using Basket.Interfaces.Repositories;
+using Basket.Models;
+
+namespace Basket.Interfaces.Services;
+
+public class BasketItemUpdater : BackgroundService
```
Comment (krylatij) [inline Homework4/AzureServiceBusMessageBroker/Basket Service/Interfaces/Services/BasketItemUpdater.cs]: why it is inside Interfaces?

File: Homework4/AzureServiceBusMessageBroker/Basket Service/Interfaces/Services/BasketItemUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        try
+        {
+            await _listnerService.ListenMessagesAsync<BasketItemUpdateMessage>(HandleBasketItemUpdateMessage, HandleBasketItemUpdateErrorMessage);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex.Message);
```
Comment (krylatij) [inline Homework4/AzureServiceBusMessageBroker/Basket Service/Interfaces/Services/BasketItemUpdater.cs]: always log the full exception

**Review by krylatij — CHANGES_REQUESTED** [review]
please re-organize services/interfaces and adjust exception handling, don't catch exceptions if you don't want to process them

File: Homework4/AzureServiceBusMessageBroker/Basket Service/Interfaces/Services/BasketItemUpdater.cs (hunk 1 above)
Comment (SonaArshakyan) [reply Homework4/AzureServiceBusMessageBroker/Basket Service/Interfaces/Services/BasketItemUpdater.cs]: changed Initial logic and forgot to replace it

**Pull request merged**